### ISSUE TITLE
Unknown or bad timezone / Add support for windows timezones

### ISSUE THREAD

**Issue by kbarendrecht** [rlanvin/php-rrule#69]
`Exception: DateTimeZone::__construct(): Unknown or bad timezone (W. Europe Standard Time) in /****/vendor/rlanvin/php-rrule/src/RfcParser.php:131`

Upon further inspection it seems that this timezone is a creation by [Microsoft](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/default-time-zones). This format is used by Microsoft's mail services (Windows mail, Outlook). It would be nice if a simple translation from windows timezone to Php timezone was added.

Something as such would suffice:
https://github.com/sabre-io/vobject/blob/master/lib/timezonedata/windowszones.php

**Comment by rlanvin** [comment]
Hello, thanks for reporting this. What are you trying to do exactly? Could you provide the code that triggers this exception?

To be honest, I'm a bit reluctant to include specific code for non-standard timezones, so let's see what is your use case.

**Comment by kbarendrecht** [comment]
The error occurerd after tyring to parse a RRule from a Microsoft Exchange Calender iCal export.
`/****/vendor/rlanvin/php-rrule/src/RRule.php(199): RRule\RfcParser::parseRRule('DTSTART;TZID=W. Europe Standard Time', NULL)`

I was already doubting if you would want to add it in this project, since it is very clean right now. 😄  I can add the translation from MS-timezone to Php-timezone in my own project aswell. I did want to let you know though, since it is used by Microsoft, and used a lot, more people might want this.

**Comment by rlanvin** [comment]
@kbarendrecht I've been thinking about it a bit, and honestly I can't believe Microsoft is still doing its thing of using non-standard crap... That being said, do you know if such RRule can be imported, e.g. in Google Calendar? I'd say, if it's not supported there, then I will not care. If it's supported by other 3rd party tools, I might also just add it to the lib... I'm also happy to get more feedback from everyone who would have the same issue.

**Comment by kbarendrecht** [comment]
I just tried it, and a .ics file exported form Microsoft Outlook can be imported with Google Calendar.
The .ics file has rules such as `TZID:W. Europe Standard Time`, `DTSTART;TZID="W. Europe Standard Time":20180802T110000`, `DTEND;TZID="W. Europe Standard Time":20180802T113000`.

Also found out that it's even allowed by the [iCal specification](https://www.kanzaki.com/docs/ical/tzid.html). It does not require a default for the timezones, just suggests the usage of Olson's timezone database.

**Comment by rlanvin** [comment]
@kbarendrecht I added support in master. It's somewhat basic for now - based on what I saw in Sabre project, it seems there are a LOT more special cases - but it should cover your use case. Feel free to test (using `dev-master`  in composer) and let me know your feedback.

**Comment by kbarendrecht** [comment]
Most of the iCals I tested from Exchange are parsed correctly but found some still throw an error:
`InvalidArgumentException: Invalid UNTIL property: The value of the UNTIL rule part MUST be a date if DTSTART is a date. in /****/vendor/rlanvin/php-rrule/src/RfcParser.php:166`.


A VEVENT from on of the faulty exports from Exchange:
```
BEGIN:VEVENT
DESCRIPTION:\n\n
RRULE:FREQ=YEARLY;UNTIL=20191231T230000Z;INTERVAL=1;BYMONTHDAY=1;BYMONTH=1
SUMMARY:****
DTSTART;VALUE=DATE:20200101
DTEND;VALUE=DATE:20200102
UID:040000008200E00074C5B7101A82E00800000000400EA87920FAD201000000000000000
 010000000174A06672567FE4CB31A7E0909D8F5B2
CLASS:PUBLIC
PRIORITY:5
DTSTAMP:20191010T131152Z
TRANSP:OPAQUE
STATUS:CONFIRMED
SEQUENCE:1
LOCATION:
X-MICROSOFT-CDO-APPT-SEQUENCE:1
X-MICROSOFT-CDO-BUSYSTATUS:FREE
X-MICROSOFT-CDO-INTENDEDSTATUS:BUSY
X-MICROSOFT-CDO-ALLDAYEVENT:TRUE
X-MICROSOFT-CDO-IMPORTANCE:1
X-MICROSOFT-CDO-INSTTYPE:1
X-MICROSOFT-DISALLOW-COUNTER:FALSE
END:VEVENT
```


This shouldn't be "fixed" though since php-rrule has implemented the spec correctly on this, unlike Microsoft's Exchange / Outlook Web App. 👍 

Thank you for your efforts on this!

**Comment by rlanvin** [comment]
@kbarendrecht I released v2.2.0 with support for Windows Timezone. For the other problem you are mentioning, yeah it's basically not compliant with the specs... I guess the approach here would be to relax the validation, but I'm not a big fan as this opens the door to undefined behaviors (there's is a reason those are not part of the RFC I guess). Thanks for reporting this bug!
